code/modules/mining/mine_turfs.dm

```diff
@@ -692,10 +692,14 @@ var/global/list/rockTurfEdgeCache
 	temperature = 300
 	baseturf = /turf/simulated/chasm/straight_down/lava_land_surface
 
+/turf/simulated/chasm/straight_down/lava_land_surface/drop(atom/movable/AM)
+	visible_message("[AM] falls into [src]!")
+	qdel(AM)
```

Comment:
I'l let you know when I have time to code another Z level.

Also the death isn't a binary "you have a suit/don't have a suit," it's a choice to go for the loot or run to safety, factoring in how close you are (did you kill it with a gun? Or get up close), how many mobs are about/how they might impede you, how fast you can still run, etc.